takealot.user.js

```diff
@@ -0,0 +1,563 @@
+						LOGGER.info("Disc itterate: ", k);
+
+						// Tracks
+						var tracklistarray = new Array(); // create the track list array
+
+						for (var j = 0; j < alltracklist.length - 1; j++) { // changed j to 0 and length-1 as Artist is at end
+							// do regex here and if disc = k then, so another if loop
+							var trackdetails = alltracklist[j].textContent.trim();
+							disctracktitleregex = /\[ Disc (\d{2}) Track.(\b\d{2}) \] (.*)/;
+							var disctracktitle = trackdetails.match(disctracktitleregex);
+
+							var currentdiscnumber = parseInt(disctracktitle[1]);
+
+							if (currentdiscnumber == k) {
+
+								var track = new Object();
+								track.number = parseInt(disctracktitle[2]);
+								track.title = disctracktitle[3];
+								LOGGER.debug("The track object: ", currentdiscnumber + ' - ' + track.number + " - " + track.title);
+
+								tracklistarray.push(track);
+							}
+						}
+						disclistarray.push(tracklistarray);
+					}
+
+					LOGGER.debug("** Disclist Array *** ", disclistarray);
+					//LOGGER.debug("** Tracklist Array *** ", tracklistarray);
+					break;
+			}
+		}
+
+	}
+
+	// Logic added to derive the release title from the heading if missing from product info
+	if (releasetitle == "") {
+		// sample header "Huisgenoot Se 20 Country-Treffers - Various Artists (CD)"
+		var MediaHeading = document.querySelectorAll("h1.fn");
+		// LOGGER.debug(MediaHeading[0].innerText);
+		var TitleStr = MediaHeading[0].innerText;
+		var TitleRegex = /(.*)-(.*)+\s\(([^)]+)\)/;
+		var HeadArray = TitleStr.match(TitleRegex);
+
+		if (HeadArray[1].trim() == releaseartist) {
+			LOGGER.debug('matched title equal releaseartist therefore swapped');
+			releasetitle = HeadArray[2].trim();
+		} else {
+			LOGGER.debug('not swapped');
+			releasetitle = HeadArray[1].trim();
+		}
+		LOGGER.debug("The value is :", releasetitle);
+		// therefore if title = artist move to next item in array ... a real hack
+	}
+
+	release = new Object();
+
+	// Release artist credit
+	release.artist_credit = new Array();
+
+	var artist_name = releaseartist;
+	release.artist_credit.push({
+		'artist_name': artist_name
+	});
+
+	// Release title
+	release.title = releasetitle;
+	// Release Barcode
+	release.barcode = releasebarcode;
+
+	// Default status is official
+	release.status = 'official';
+
+	// Other hard-coded info
+	release.script = 'Latn';
+
+	release.country = Countries[releasecountry];
+	release.language = Languages[releaselanguage];
+
+
+
+	release.discs = new Array();
+	for (var l = 0; l < lastdiscnumber; l++) {
+		LOGGER.debug("*** Disc position ***", l + 1);
+		LOGGER.debug("*** The Disk tracks *** ", disclistarray[l]);
+		var disc = {
+			'position': l + 1,
+			'tracks': disclistarray[l]
+		};
+		release.discs.push(disc);
+	}
+
+	release.labels = [];
+
+	// Release URL
+	release.urls = new Array();
+	release.urls.push({
+		'url': window.location.href,
+		'link_type': 74
```

Comment:
You should use `MBImport.URL_TYPES.purchase_for_download` here instead of the numerical value.